src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.KeyTrans.cs

```diff
@@ -96,7 +96,7 @@ internal ManagedKeyTransPal(KeyTransRecipientInfoAsn asn)
                 }
                 else
                 {
-                    using (RSA rsa = cert.GetRSAPrivateKey())
+                    using (RSA? rsa = cert!.GetRSAPrivateKey())
```

Comment:
What prevents `cert` from being null here?  The method advertises that it allows for a null cert to be passed in.

Reply:
`privateKey` or `cert` should be non null here, at this point `privateKey` was null that means cert was not null

Reply:
That's a guarantee made by all callers?  I'm not seeing anything above that guarantees that.  Do we need an Assert to that effect (it wouldn't help with the warning suppression, but it would make it clear to a reader like me)?

Reply:
Yes there is nothing implies that within this method, but there is an assert in calling methods:
https://github.com/dotnet/runtime/blob/16d04a3932bb16f5ff211ee9fbea377dd258d54f/src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/DecryptorPalWindows.Decrypt.cs#L28

Do you prefer to have additional assertion here?

Reply:
> Do you prefer to have additional assertion here?

I would prefer that, since this method is relying on that condition being true.  @bartonjs, that ok with you?

Reply:
Yeah, I almost suggested adding the assert since I had to track it back myself; then decided I was being overly picky and didn't :).